toolkit/tools/internal/safechroot/safechroot.go

```diff
@@ -262,7 +290,13 @@ func (c *Chroot) AddFiles(filesToCopy ...FileToCopy) (err error) {
 	for _, f := range filesToCopy {
 		dest := filepath.Join(c.rootDir, f.Dest)
 		logger.Log.Debugf("Copying '%s' to worker '%s'", f.Src, dest)
-		err = file.Copy(f.Src, dest)
+
+		if f.FileMode != nil {
```

Comment:
Setting the file permissions of AdditionalFiles is a missing feature of the Mariner Toolkit.